Updates to the MYNN surface layer scheme

joeolson42 opened the pull request:
This update includes the following fixes/mods: 
1) bug fix for init routine - was not initializing the psi functions properly due to missing errmsg and errflg arguments (this required mods to the surface wrapper and the associated meta file), 
2) bug fix for using initialized variable (u* from previous time step) when grid points with 100% ice begins to melt or when ice begins to develop on a grid which previously had 0% ice. 
3) reduced overly liberal range of allowable z/L.
4) made a change to not use the final indice (1000) in the psi tables. This is an unsolved mystery (from NCAR/MMM) as to why it causes a crash (in the WRF framework), but we should keep the code consistent between CCPP and WRF as much as possible.

Associated PRs:

https://github.com/NOAA-GSL/ccpp-physics/pull/67
https://github.com/NOAA-GSL/fv3atm/pull/63
https://github.com/NOAA-GSL/ufs-weather-model/pull/52

For regression testing, see https://github.com/NOAA-GSL/ufs-weather-model/pull/52.

joeolson42 committed:
Merge pull request #5 from NOAA-GSD/gsd/develop

updating fork

joeolson42 committed:
Merge pull request #6 from NOAA-GSD/gsd/develop

updating fork

DomHeinzeller committed:
Merge pull request #62 from climbfuji/update_gsd_develop_from_master_20201118

Update gsd/develop from master 2020/11/18

joeolson42 committed:
Merge pull request #7 from NOAA-GSL/gsd/develop

update branch

joeolson42 committed:
Updates to mynn sfc layer: 1) bugfix for init routine, 2) bugfix for using unitialized variable when 100% ice melts or ice develops on a grid with 0% ice, 3) reduced overly liberal range of allowable z/L.

DomHeinzeller reviewed (commented):
These code changes look good to me. I am going to test this now with the noaa-gsl ufs-weather-model and create the corresponding PRs for fv3atm and ufs-weather-model.

Will approve after testing is complete.

DomHeinzeller reviewed (commented):
@joeolson42 update - your PR is not up to date. Can you please check that you have pulled in the latest code from https://github.com/noaa-gsl/ccpp-physics, branch gsl/develop ? And then push the update to your branch. Thanks.

joeolson42 commented:
I updated my fork yesterday, but perhaps I updated it from gsd/develop
instead of gsl/develop by accident???

On Tue, Dec 15, 2020 at 2:47 PM DomHeinzeller <notifications@github.com>
wrote:

> *@DomHeinzeller* commented on this pull request.
>
> @joeolson42 <https://github.com/joeolson42> update - your PR is not up to
> date. Can you please check that you have pulled in the latest code from
> https://github.com/noaa-gsl/ccpp-physics, branch gsl/develop ? And then
> push the update to your branch. Thanks.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-GSL/ccpp-physics/pull/67#pullrequestreview-552935067>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADLRR3XRJRSDND5BX3RWTK3SU7KP7ANCNFSM4U3OWF4Q>
> .
>


-- 
Joseph Olson
Meteorologist
Environmental Prediction Advancement Division
Global Systems Laboratory
NOAA-Earth System Research Labs
Boulder, Colorado

DomHeinzeller commented:
Yes, that's what it looks like.

> On Dec 15, 2020, at 3:00 PM, Joseph Olson <notifications@github.com> wrote:
> 
> 
> I updated my fork yesterday, but perhaps I updated it from gsd/develop
> instead of gsl/develop by accident???
> 
> On Tue, Dec 15, 2020 at 2:47 PM DomHeinzeller <notifications@github.com>
> wrote:
> 
> > *@DomHeinzeller* commented on this pull request.
> >
> > @joeolson42 <https://github.com/joeolson42> update - your PR is not up to
> > date. Can you please check that you have pulled in the latest code from
> > https://github.com/noaa-gsl/ccpp-physics, branch gsl/develop ? And then
> > push the update to your branch. Thanks.
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/NOAA-GSL/ccpp-physics/pull/67#pullrequestreview-552935067>,
> > or unsubscribe
> > <https://github.com/notifications/unsubscribe-auth/ADLRR3XRJRSDND5BX3RWTK3SU7KP7ANCNFSM4U3OWF4Q>
> > .
> >
> 
> 
> -- 
> Joseph Olson
> Meteorologist
> Environmental Prediction Advancement Division
> Global Systems Laboratory
> NOAA-Earth System Research Labs
> Boulder, Colorado
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/NOAA-GSL/ccpp-physics/pull/67#issuecomment-745594500>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AN7FF5H5XB4SR5IC57CPIA3SU7MAFANCNFSM4U3OWF4Q>.
>

joeolson42 committed:
Merge pull request #8 from NOAA-GSL/gsl/develop

updating from gsl/develop instead of gsd/develop

DomHeinzeller commented:
> Yes, that's what it looks like.
> […](#)
> On Dec 15, 2020, at 3:00 PM, Joseph Olson ***@***.***> wrote: I updated my fork yesterday, but perhaps I updated it from gsd/develop instead of gsl/develop by accident??? On Tue, Dec 15, 2020 at 2:47 PM DomHeinzeller ***@***.***> wrote: > *@DomHeinzeller* commented on this pull request. > > @joeolson42 <https://github.com/joeolson42> update - your PR is not up to > date. Can you please check that you have pulled in the latest code from > https://github.com/noaa-gsl/ccpp-physics, branch gsl/develop ? And then > push the update to your branch. Thanks. > > — > You are receiving this because you were mentioned. > Reply to this email directly, view it on GitHub > <[#67 (review)](https://github.com/NOAA-GSL/ccpp-physics/pull/67#pullrequestreview-552935067)>, > or unsubscribe > <https://github.com/notifications/unsubscribe-auth/ADLRR3XRJRSDND5BX3RWTK3SU7KP7ANCNFSM4U3OWF4Q> > . > -- Joseph Olson Meteorologist Environmental Prediction Advancement Division Global Systems Laboratory NOAA-Earth System Research Labs Boulder, Colorado — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <[#67 (comment)](https://github.com/NOAA-GSL/ccpp-physics/pull/67#issuecomment-745594500)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AN7FF5H5XB4SR5IC57CPIA3SU7MAFANCNFSM4U3OWF4Q>.

Ok, this is better. Will start testing.

DomHeinzeller commented:
@joeolson42 quick update, the regression tests look good, I'll update the ufs-weather-model PR with a few details in a minute and then merge the code.
